Dockerfile

```diff
@@ -1,52 +1,53 @@
-# Build the manager binary
-FROM golang:1.22 as builder
+FROM nixos/nix:latest AS go-builder
+COPY Makefile Makefile
+COPY shell.nix shell.nix
 
+# Set the working directory
 WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
-RUN go install golang.org/x/tools/cmd/goimports@latest
```

Comment:
Maybe these can be removed. Not sure about the `go mod download` though, that is just getting the deps from the current go module definitions, not all needs to go though nix.